File: agni-core/src/test/kotlin/dev/agni/core/protocol/ResponseTest.kt
package dev.agni.core.protocol

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ResponseTest {
    private fun bytesOf(response: Response): String = response.toBytes().toString(Charsets.UTF_8)

    @Test
    fun `pong encodes to PONG`() {
        assertEquals("PONG", bytesOf(Response.Pong))
    }

    @Test
    fun `ok encodes to OK`() {
        assertEquals("OK", bytesOf(Response.Ok))
    }

    @Test
    fun `value encodes to raw bytes`() {
        assertEquals("hello", bytesOf(Response.Value("hello".toByteArray())))
    }

    @Test
    fun `null encodes to NULL`() {
        assertEquals("NULL", bytesOf(Response.Null))
    }

    @Test
    fun `error encodes with ERR prefix`() {
        assertEquals("ERR bad command", bytesOf(Response.Error("bad command")))
    }
}

File: protocol/response_test.go
package protocol

import (
	"bytes"
	"testing"
)

func TestPongBytes(t *testing.T) {
	if got := (Response{Kind: Pong}).Bytes(); !bytes.Equal(got, []byte("PONG")) {
		t.Fatalf("got %q, want PONG", got)
	}
}

func TestValueBytesReturnsRawValue(t *testing.T) {
	r := Response{Kind: Value, Value: []byte("hello")}
	if got := r.Bytes(); !bytes.Equal(got, []byte("hello")) {
		t.Fatalf("got %q, want hello", got)
	}
}

func TestErrorBytesIncludesMessage(t *testing.T) {
	r := Response{Kind: Error, Message: "unknown command 'FOO'"}
	if got := string(r.Bytes()); got != "ERR unknown command 'FOO'" {
		t.Fatalf("got %q", got)
	}
}
